Bio/SeqIO/QualityIO.py

```diff
@@ -2112,7 +2145,15 @@ def _fastq_generic(
             # map the qual...
             qual = old_qual.translate(mapping)
             if null in qual:
-                raise ValueError("Invalid character in quality string")
+                invalid_index = qual.find(null)
+                raise ValueError(
+                    _invalid_char_msg(
+                        invalid_index,
```

Comment:
Perhaps we should define an `InvalidCharError` that derives from `ValueError`.
Then invalid_index and full_string can be attributes of instances of `InvalidCharError`, and make them accessible to the user.
The function `_invalid_char_msg` would then be the `__str__` method, which is a bit cleaner.

Reply:
The specific exception does seem quite elegant 👍

Reply:
Do you think I should put it in the project __init__.py file? I'm not sure I'm a fan of the long `Bio.SeqIO.QualityIO.InvalidCharError` name displayed in the error message

Reply:
OK, fair enough. I thought about compromising and setting `__module__` to `Bio.SeqIO`, but decided I'd rather avoid spoofing the source module. We can always move it up a level later if desired